editor/store/reducer.js

```diff
@@ -236,6 +236,30 @@ export const editor = flow( [
 				];
 			}
 
+			case 'MOVE_BLOCK_TO_INDEX': {
```

Comment:
Can we and do you think it's worth it to combine `MOVE_BLOCK_TO_INDEX `, `MOVE_BLOCKS_UP `, and `MOVE_BLOCKS_DOWN `?

Reply:
I totally think it's worth it. I don't see why we can't do that, but I need to familiarise myself with how the other cases are called. I believe a single `MOVE_BLOCKS_TO_INDEX` would be ideal, and it could go in a separate PR as well.

Reply:
This is actually harder to pull off right than it seems on surface. It's easier when you are moving a single block, but with multiple blocks it becomes cognitively harder to visualise the list first without the blocks (in order to identify the index position to move them to). I suspect this may be the reason the devs chose to go with two separate, simpler functions here.

I have been experimenting with a couple of things, but I constantly find it hard to communicate the behaviour. My best "single function" approach is to have different behaviour when moving a selection of blocks to a lower index than to a higher one (in the former case you would end up moving the blocks before the current item at that index, and in the latter after it). I fear some may find this too confusing to appreciate. 

However, we certainly need to improve on the `MOVE_BLOCKS_UP` and `MOVE_BLOCKS_DOWN` functions, as their functionality is quite limited as it stands (they only serve to move blocks to the previous or next index). My thinking now goes to introducing a replacement for each a la `MOVE_BLOCKS_BEFORE_INDEX` and `MOVE_BLOCKS_AFTER_INDEX`, with each accepting an index position on which to move the blocks before or after. This would be easier to communicate and clearer on what the expected and underlying behaviour is. What do you think? Am I giving this too much thought? Happy to jump on a call and hack something together if you like 😄

Reply:
Fair enough, fine leaving it as it is.